src/LibraryManager/CacheService/CacheService.cs

```diff
@@ -0,0 +1,128 @@
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Web.LibraryManager.Contracts;
+
+namespace Microsoft.Web.LibraryManager
+{
+    /// <summary>
+    /// Service to manage basic operations on libraries cache
+    /// </summary>
+    internal class CacheService
+    {
+        // TO DO: Move these expirations to the provider 
+        private readonly int _catalogExpiresAfterDays = 3;
+        private readonly int _libraryExpiresAfterDays = 1;
+        private IWebRequestHandler _requestHandler;
+
+        public CacheService(IWebRequestHandler requestHandler)
+        {
+            _requestHandler = requestHandler;
+        }
+
+        /// <summary>
+        /// Deletes files from cache 
+        /// </summary>
+        public void ClearCache()
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Returns the provider's catalog from the provided Url to cacheFile
+        /// </summary>
+        /// <param name="url"></param>
+        /// <param name="cacheFile"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task<string> GetCatalogAsync(string url, string cacheFile, CancellationToken cancellationToken)
+        {
+            return await GetResourceAsync(url, cacheFile, _catalogExpiresAfterDays, cancellationToken);
+        }
+
+        /// <summary>
+        /// Returns library metadata from provided Url to cacheFile
+        /// </summary>
+        /// <param name="url"></param>
+        /// <param name="cacheFile"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+
+        public async Task<string> GetMetadataAsync(string url, string cacheFile, CancellationToken cancellationToken)
+        {
+            return await GetResourceAsync(url, cacheFile, _libraryExpiresAfterDays, cancellationToken);
+        }
+
+        /// <summary>
+        /// Retruns the library content from the provider url and writes to cacheFile
+        /// </summary>
+        /// <param name="url"></param>
+        /// <param name="cacheFile"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task<string> GetLibraryAsync(string url, string cacheFile, CancellationToken cancellationToken)
+        {
+            return await GetResourceAsync(url, cacheFile, _libraryExpiresAfterDays, cancellationToken);
+        }
+
+        /// <summary>
+        /// Downloads a resource from specified url to a destination file
+        /// </summary>
+        /// <param name="url"></param>
+        /// <param name="fileName"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        private async Task DownloadToFileAsync(string url, string fileName, CancellationToken cancellationToken)
```

Comment:
I assume this is not intended to work with filesystem provider?

Reply:
No, FileSystemProvider does not have a cache